runtime/doc/map.txt

```diff
@@ -1430,6 +1430,145 @@ Possible values are (second column is the short name used in listing):
     -addr=other		  ?	other kind of range
 
 
+Incremental preview ~
+                                                  *:command-preview* {nvim-api}
+It's possible to allow commands to be incrementally previewed when
+'inccommand' is enabled by using the preview command flag. The preview
+flag may only be used through the Lua API (see |nvim_create_user_command()|).
+
+The preview callback must be a Lua function. When command preview is
+triggered, the preview callback will be called with 3 arguments, an `opts` table
+which is the same as the table passed to |nvim_create_user_command()| callbacks,
+the preview namespace number that's used to set highlights for the preview,
+and the preview buffer number that's used for the preview window when
+`inccommand=split`. Note that if `inccommand=nosplit`, the preview buffer number
+will be `nil`.
+
+In order for command preview to be shown, these steps need to be followed:
```

Comment:
- This documentation is a bit verbose. I think it can be reduced.
- Describing code in english is not really helpful. Just show a function signature.
    - I intentionally avoid backticks because overuse of backticks can be distracting in vimdocs.

Here's a suggestion for this part:

```suggestion
The preview callback must be a Lua function with this signature: >

    function cmdpreview(opts, ns, buf)

where "opts" has the same form as that given to |nvim_create_user_command()|
callbacks, "ns" is the preview namespace id for highlights, and "buf" is the
buffer where your preview routine will directly modify to show the previewed
results (for "inccommand=split", or nil for  "inccommand=nosplit").

Your command preview routine must implement this protocol:
```